File: Breast Cancer Logistic Regression.R
#Import Dataset into Dataframe:
DataSet <- read.csv("C:\\Users\\abdul\\Desktop\\R Programming\\Freestyle\\Breast Cancer\\Dataset\\breast_cancer.csv")
DataSet[DataSet$Class == 2, "Class"] <- 0
DataSet[DataSet$Class == 4, "Class"] <- 1

#Splitting the Dataset into a Training Set and a Data Set:
set.seed(0)
split <- sample(c(rep(0, 0.8*nrow(DataSet)),
                rep(1, 0.2*nrow(DataSet))))
train <- DataSet[split == 0,]
test <- DataSet[split == 1,]

#Training the Logistic Regression Model (on the training set):
model <- glm(Class ~ .-Sample.code.number, data = train, family = binomial)

#Predicting the Test Set Results:
prediction <- predict(model, newdata = test, type = "response")

#Making a Confusion Matrix:
pred.model <- ifelse(prediction>0.5, "Malignant", "Benign")
test$Class <- ifelse(test$Class == 1, "Malignant", "Benign")
cm <- table(pred.model, test$Class)
accuracy <- mean(pred.model == test$Class)

#Computing the Accuracy with k-Fold Cross Validation:
#I do not know how to do this in R. 

#Summary:
Summary <- list("Coefficients" = summary(model)$coef, "Confusion Matrix" = cm, "Accuracy" = accuracy)
Summary

File: Breast Cancer Logistic Regression.py
#Import Pandas Package:
import pandas as pd

#Import Dataset:
dataset = pd.read_csv('C:\\Users\\abdul\\Desktop\\Final Folder\\Dataset\\breast_cancer.csv')
#Matrix of Features/Independent Variables:
X = dataset.iloc[ : , 1:-1].values
#Dependent-Variable Vector:
y = dataset.iloc[ : , -1].values

#Splitting the Dataset into a Training Set and a Data Set:
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 0)

#Training the Logistic Regression Model (on the training set):
from sklearn.linear_model import LogisticRegression
classifier = LogisticRegression(random_state = 0)
classifier.fit(X_train, y_train)

#Predicting the Test Set Results:
y_pred = classifier.predict(x_test)

#Making a Confusion Matrix:
from sklearn.metrics import confusion_matrix
cm = confusion_matrix(y_test, y_pred)
print(cm)

#Computing the Accuracy with k-Fold Cross Validation:

from sklearn.model_selection import cross_val_score
